components/mpas-ocean/driver/ECDA_mpaso_obs_mod.F

```diff
@@ -0,0 +1,636 @@
+    character(len=var_name_len)          ::   uncert_name='null'    ! the obs variable name in ncfile 
+    character(len=var_name_len)          ::   lonname='longitude'           ! the lon coordinator name in ncfile 
+    character(len=var_name_len)          ::   latname='latitude'            ! the lat coordinator name in ncfile 
+    character(len=var_name_len)          ::   vertical_name= 'depth'         ! vertical
+    character(len=var_name_len)          ::   dm_name= 'records'
+    character(len=10)                    ::   init_obs_YMDH='2010010112'     ! the inital time to observations
+    integer                              ::   uncert_type = 1       ! variance 0 ,std 1 default, not valid when err_var0 >0 applies
+    integer                              ::   grid_type = 1      !  netcdf file, 1,2 are grid file, 1: model grid, 2 other resolution, need read coordinator information, 3: station data, (lon,lat,lev, value,error) 
+    integer                              ::   obs_dm = 2                                ! data dimension in file, only used for grid file
+    integer                              ::   start_layer = 1                           ! data start  from (vertical layer) for 3D,only used for grid file
+    integer                              ::   nlayer                                 ! total vertical layer for 3D,only used for grid file
+    integer                              ::   ilayer                                 ! pick vertical layer ,only used for grid file
+    integer                              ::   inst_type =0                             ! instrument types are defined by platform class (e.g. MOORING, DROP) and instrument type (XBT, CTD, ...), not active now @YL
+    logical                              ::   required_states(n_OCN_state)=.false.   ! the required state variables for obsope 
+    logical                              ::   update_states(n_OCN_state)=.false.     ! the updated  state variables for current variable 
+    real(r8)                             ::   skip_sfc(n_OCN_state)=0                ! skip surface for ssh lik observation 
+    real(r8)                             ::   RR_xy=10.0, RR_z=200.0                 !
+    real(r8)                             ::   Mx_PO_dist=10.0                        ! maxium distance allowed between observation and forecast
+    real(r8)                             ::   Mx_misfit=25                           ! maxium rate allowed for inflate  observation uncertainty
+    real(r8)                             ::   err_var0 = -1.0                          ! specify constant observation error variance (>0). When <=0, input from observation file, default.
+    real(r8)                             ::   adj_scale= 1.0               !some variable need adjust scale
+
+    integer :: ierr
+    integer :: iunit
+    real(r8):: obs_err_var0
+    namelist /obs_assim_nml/obs_name,obs_type,filename,inst_type,varname,lonname,latname,vertical_name,&
```

Comment:
is this in the same name list as mpas-ocean?  i think it would be nice to have that together with a flag to enable/disable the capability

Reply:
There are two namelists. The one inputs the observation types. Each observation type has its own namelist to specify the filename, longitude name , latitude name, instrument type ...